src/main/java/seedu/canoe/logic/commands/UnmarkAttendanceCommand.java

```diff
@@ -0,0 +1,127 @@
+import static seedu.canoe.model.Model.PREDICATE_SHOW_ALL_STUDENTS;
+
+import java.util.List;
+import java.util.logging.Logger;
+
+import seedu.canoe.commons.core.LogsCenter;
+import seedu.canoe.commons.core.index.Index;
+import seedu.canoe.logic.commands.exceptions.CommandException;
+import seedu.canoe.model.Model;
+import seedu.canoe.model.student.AnyMatchPredicateList;
+import seedu.canoe.model.student.Attendance;
+import seedu.canoe.model.student.Student;
+import seedu.canoe.model.training.Training;
+
+public class UnmarkAttendanceCommand extends Command {
+
+    public static final Logger LOGGER = LogsCenter.getLogger(UnmarkAttendanceCommand.class);
+
+    public static final String COMMAND_WORD = "unmark-attendance";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Gets all students in the training session"
+            + " whose Ids corresponds to the specified Ids and unmark them as attended the training session.\n"
+            + "Parameters: ID [MORE_IDS]...\n"
+            + "Example: " + COMMAND_WORD + " 2 id/1,4,19";
+
+    public static final String MESSAGE_NO_STUDENTS_SPECIFIED = "At least one student to be added must be specified.";
+    public static final String MESSAGE_INVALID_STUDENT_MARKED = "Some students do not have specified"
+            + " training session scheduled!";
+    public static final String MESSAGE_MARK_AS_ATTENDED_SUCCESS = "Unmarked these students as attended: %1$s!";
+
+    private final Index trainingIndex;
+    private final AnyMatchPredicateList predicates;
+
+    /**
+     * Constructs a new UnmarkAttendanceCommand object.
+     *
+     * @param trainingIndex index of training.
+     * @param predicates list of predicates.
+     */
+    public UnmarkAttendanceCommand(Index trainingIndex, AnyMatchPredicateList predicates) {
+        this.trainingIndex = trainingIndex;
+        this.predicates = predicates;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        LOGGER.info("=============================[ Executing UnmarkAttendanceCommand ]===========================");
+        requireNonNull(model);
+        List<Training> lastShownList = model.getFilteredTrainingList();
+
+        if (trainingIndex.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(MESSAGE_TRAININGS_NOT_FOUND);
+        }
+
+        model.updateFilteredStudentList(predicates);
+
+        if (model.getFilteredStudentList().isEmpty()) {
+            LOGGER.warning("Ids match zero students.");
+            model.updateFilteredStudentList(PREDICATE_SHOW_ALL_STUDENTS);
+            throw new CommandException(MESSAGE_STUDENTS_NOT_FOUND);
+        }
+
+        List<Student> attendedStudents = model.getFilteredStudentList();
+
+        Training training = lastShownList.get(trainingIndex.getZeroBased());
+
+        Attendance unattendedAttendance = new Attendance(training.getDateTime());
```

Comment:
Do we want to standadise **unmarked**Attendance as the word to use instead of unattend?